contents/docker/Docker.md

```diff
@@ -0,0 +1,161 @@
+- **Lightweight footprint and minimal overhead** – Docker images are typically very small, which facilitates rapid delivery and reduces the time to deploy new application containers.
+- **Simplified maintenance** – Docker reduces effort and risk of problems with application dependencies.
+
+Let us take a detailed look at some of the above advantages.
+
+#### 2.1 Docker is lightweight 
+ 
+As we have discussed, Docker uses makes use of Containers that share the host
+system's kernel with other containers. It runs a discrete process, taking no more memory than any other executable, making it lightweight.  
+
+Unlike virtual machines, containers do not have the high overhead and hence enable more efficient usage of the underlying system and resources.
+
+
+#### 2.2 Docker allows for sharing and reuse
+
+Docker is a service that is designed to automate and deploy containers. When you
+run a container, you will start out with a base docker image. A docker
+image is a file, comprised of multiple layers, used to execute code in a Docker 
+container. 
+
+When you update configurations or install applications in your container, the
+corresponding docker image gets updated as well. This image can then be shared
+across multiple machines by different people! This way you dont have to
+repeatedly install the same software or reconfigure a container from scratch on
+a new machine.
+
+On top of this, there is this website known as the Docker hub:
+https://hub.docker.com/. This is basically a collection of over 100,000 docker
+images uploaded by the community. You can use this website to load in a base
+image with all the configurations you already need. You can also share your
+custom images on this site for others!
+
+#### 2.3 Docker is easy to get started with 
+
+Despite its complicated architecture, it is not difficult for a complete beginner to get started with Docker.
+We have covered how docker hub provides various images for someone to use. On
+top of this, Docker also provides in-built automation of deployment through what
+is known as a docker file. 
+
+A simple example of a docker file is as follows:
+```
+# our base image
+FROM python:3-onbuild
+
+# specify the port number the container should expose
+EXPOSE 5000
+
+# run the application
+CMD ["python", "./app.py"]
+```
+
+This file helps to load in a onbuild python3 image - an onbuild image is a
+specific image that is able to copy an existing server and run it automatically.
+The docker file also allows bash commands to be run as you can see from the last
+line above.
+
+There are plenty of guides out there that show you the extended functionality a
+docker file can provide. Even if you are a complete beginner, it is not
+difficult to run an Docker file that someone else has built. Many of the docker
+hub images also provide sample docker file builds!
```

Comment:
Actually I think docker-hub community is the key reason why docker is so easy to get started with, because in most of the cases, you do not need to build your own image! And even if you do, you can just extend existing images. I feel this point should be emphasised.